nni/nas/oneshot/pytorch/profiler.py

```diff
@@ -0,0 +1,221 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Guide the one-shot strategy to sample architecture within a target latency.
+
+This module converts the profiling results returned by profiler to something
+that one-shot strategies can understand. For example, a loss or some penalty to the reward.
+
+This file is experimentally placed in the oneshot package.
+It might be moved to a more general place in the future.
+"""
+
+from __future__ import annotations
+
+import logging
+from typing_extensions import Literal
+
+import numpy as np
+import torch
+from torch import nn
+
+from nni.mutable import Sample
+from nni.nas.profiler import Profiler, ExpressionProfiler
+
+from .supermodule._expression_utils import expression_expectation
```

Comment:
Why underscore prefixed?

Reply:
Experimental utils. Don't want to expose publicly.